x-compile for Android/MIPS fails

Original [issue 109](https://code.google.com/p/leveldb/issues/detail?id=109) created by ulcica on 2012-08-03T11:51:31.000Z:

Use the android branch. Edit leveldb/jni/Application.mk

APP_ABI := mips

The ndk-build output:
-----------------------------------------
Compile++ mips   : leveldb &lt;= builder.cc
In file included from ./port/port.h:18:0,
                 from ./db/filename.h:14,
                 from jni/.././db/builder.cc:7:
./port/port_android.h:13:22: fatal error: cstdatomic: No such file or directory
compilation terminated.
make: *** [obj/local/mips/objs/leveldb/__/./db/builder.o] Error 1
-----------------------------------------

compiled using android-ndk-r8b, Mac OS X 10.7.4

--- a/port/atomic_pointer.h
+++ b/port/atomic_pointer.h
@@ -39,6 +39,8 @@
 #define ARCH_CPU_ARM64_FAMILY 1
 #elif defined(__ppc__) || defined(__powerpc__) || defined(__powerpc64__)
 #define ARCH_CPU_PPC_FAMILY 1
+#elif defined(__mips__)
+#define ARCH_CPU_MIPS_FAMILY 1
 #endif
 
 namespace leveldb {
@@ -107,6 +109,13 @@ inline void MemoryBarrier() {
   // TODO for some powerpc expert: is there a cheaper suitable variant?
   // Perhaps by having separate barriers for acquire and release ops.
   asm volatile("sync" : : : "memory");
+}
+#define LEVELDB_HAVE_MEMORY_BARRIER
+
+// MIPS
+#elif defined(ARCH_CPU_MIPS_FAMILY) && defined(__GNUC__)
+inline void MemoryBarrier() {
+  __asm__ __volatile__("sync" : : : "memory");
 }
 #define LEVELDB_HAVE_MEMORY_BARRIER
 

